src/main/kotlin/com/demonwav/mcdev/platform/mixin/config/MixinConfigImportOptimizer.kt

```diff
@@ -0,0 +1,130 @@
+/*
+ * Minecraft Dev for IntelliJ
+ *
+ * https://minecraftdev.org
+ *
+ * Copyright (c) 2018 minecraft-dev
+ *
+ * MIT License
+ */
+
+package com.demonwav.mcdev.platform.mixin.config
+
+import com.google.common.collect.Comparators
+import com.google.gson.Gson
+import com.intellij.json.psi.JsonArray
+import com.intellij.json.psi.JsonElementGenerator
+import com.intellij.json.psi.JsonFile
+import com.intellij.json.psi.JsonObject
+import com.intellij.json.psi.JsonProperty
+import com.intellij.json.psi.JsonStringLiteral
+import com.intellij.lang.ImportOptimizer
+import com.intellij.openapi.util.EmptyRunnable
+import com.intellij.psi.PsiDocumentManager
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+
+class MixinConfigImportOptimizer : ImportOptimizer {
+
+    /**
+     * Sorts a number of (qualified) class names based on their package names
+     * and the following rules:
+     *
+     *   - Each package name is compared separately
+     *   - Default packages come first
+     *   - Class names are only compared if in the same package
+     *
+     * Example:
+     *   ```
+     *   Main
+     *   com.demonwav.mcdev.Example
+     *   com.demonwav.mcdev.HelloWorld
+     *   com.demonwav.mcdev.asset.Assets
+     *   com.demonwav.mcdev.util.ActionData
+     *   ```
+     */
```

Comment:
As long as this is the sort that's expected from this feature, that's fine by me.